content/courses/advanced-ada/chapters/exceptions.rst

```diff
@@ -215,16 +215,423 @@ For example, you can activate all policies by writing:
 
 .. note::
 
-    This section was originally  published as
+    Parts of this section were originally published as
     `Gem #142 : Exception-ally <https://www.adacore.com/gems/gem-142-exceptions>`_
 
-.. admonition:: Relevant topics
+The standard Ada run-time library provides the package :ada:`Ada.Exceptions`.
+This package provides a number of services to help analyze exceptions.
 
-    - `The Package Exceptions <http://www.ada-auth.org/standards/2xrm/html/RM-11-4-1.html>`_
+Each exception is associated with a (short) message that can be set by the code
+that raises the exception, as in the following code:
 
-.. todo::
+.. code-block:: ada
 
-    Complete section!
+    raise Constraint_Error with "some message";
+
+.. admonition:: Historically
+
+    Since Ada 2005, we can use the
+    :ada:`raise Constraint_Error with "some message"` syntax.
+    In previous versions of Ada, you had to call the :ada:`Raise_Exception`
```

Comment:
Package Ada.Exceptions was only added in Ada 95; in Ada 83 there was no way to do it at all.

Reply:
Interesting, I didn't know that. Thanks for the hint!